### ISSUE TITLE
Is there a bug with prefix_host_key?

### ISSUE THREAD

**Issue by juongithub** [websupport-sk/pecl-memcache#56]
I'm having a problem with the configuration of prefix_host_key and/or session_prefix_host_key.
It appears that under certain conditions versions 4.0.3/4.0.4 (Windows, x64, TS, PHP7.3.6) ignore their settings and always add a prefix to each key. Unfortunately, I'm currently not able to reproduce these conditions, it seems to be a random behavior. When I restart PHP the problem is resolved but after few time it appears again.
I have also used different memcached servers (under linux, under windows native + cygwin).

I'm using this code to test:
```
<?php

$pool = new MemcachePool();
$pool->addServer( '127.0.0.1' );


$pool->set( 'test1', 'value1', 0, time() + 24 * 3600 );
$pool->delete( 'test1' );
$pool->set( 'test1', 'value2', 0, time() + 24 * 3600 );
$pool->delete( 'test1' );
$pool->set( 'test1', 'value3', 0, time() + 24 * 3600 );

$_SERVER['SERVER_NAME'] = '';

echo "SERVER_NAME='$_SERVER[SERVER_NAME]' -> " . $pool->get( 'test1' ) . PHP_EOL;

$_SERVER['SERVER_NAME'] = 'this.is.a.test';

echo "SERVER_NAME='$_SERVER[SERVER_NAME]' -> " . $pool->get( 'test1' ) . PHP_EOL;

$_SERVER['SERVER_NAME'] = '';

$N = 500;

echo 'testing ' . $N . ' keys...' . PHP_EOL;

$base_key = sprintf( '%x', crc32( mt_rand() ) );

$errors = 0;

for ( $i = 0; $i < $N; $i ++ ) {
	$pool->set( $base_key . ':' . $i, $i, 0, time() + 24 * 3600 );
}
for ( $i = 0; $i < $N; $i ++ ) {
	$r = $pool->get( $base_key . ':' . $i );
	if ( $r !== $i ) {
		$errors ++;
	}
}

echo "tested keys: $N - errors: $errors" . PHP_EOL;

$errors = 0;
for ( $i = 0; $i < $N; $i ++ ) {
	$pool->delete( $base_key . ':' . $i );
	$r = $pool->get( $base_key . ':' . $i );
	if ( $r === $i ) {
		$errors ++;
	}
}

echo "deleted keys: $N - errors: $errors" . PHP_EOL;

$gets = $deletes = 0;
for ( $i = 0; $i < $N; $i ++ ) {
	$pool->set( $key = md5( mt_rand() ), ( $value = md5( mt_rand() ) ), 0, time() + 24 * 3600 );
	if ( $pool->get( $key ) === $value ) {
		$gets ++;
	}
	$pool->replace( $key, '.:|-deleted-|:.', 0, time() + 24 * 3600 );
	if ( $pool->get( $key ) !== $value ) {
		$deletes ++;
	}
}

echo "checks: $N - get: $gets - delete: $deletes" . PHP_EOL;

$pool->set( 'test1', 'value4', 0, time() + 24 * 3600 );
$pool->delete( 'test1' );
$pool->set( 'test1', 'value5', 0, time() + 24 * 3600 );
$pool->delete( 'test1' );
$pool->set( 'test1', 'value6', 0, time() + 24 * 3600 );

$_SERVER['SERVER_NAME'] = '';

echo "SERVER_NAME='$_SERVER[SERVER_NAME]' -> " . $pool->get( 'test1' ) . PHP_EOL;

$_SERVER['SERVER_NAME'] = 'this.is.a.test';

echo "SERVER_NAME='$_SERVER[SERVER_NAME]' -> " . $pool->get( 'test1' ) . PHP_EOL;

```

Sometimes it returns:
```
SERVER_NAME='' -> value3
SERVER_NAME='this.is.a.test' -> value3
testing 500 keys...
tested keys: 500 - errors: 0
deleted keys: 500 - errors: 0
checks: 500 - get: 500 - delete: 500
SERVER_NAME='' -> value6
SERVER_NAME='this.is.a.test' -> value6
```

Other times something like this:
```
SERVER_NAME='' -> value3
SERVER_NAME='this.is.a.test' -> value3
testing 500 keys...
tested keys: 500 - errors: 0
deleted keys: 500 - errors: 0
checks: 500 - get: 44 - delete: 500
SERVER_NAME='' -> 
SERVER_NAME='this.is.a.test' -> 
```

These results suggest to me that there are at least 2 issues.

**Comment by juongithub** [comment]
Just to add more information, every time I see the wrong output I also see this PHP error:
Notice: MemcachePool::delete(): Server 127.0.0.1 (tcp 11211, udp 1) failed with: Network timeout (0)

**Comment by juongithub** [comment]
I think the issue rises when the code tries to reconnect to the memcached server after a timeout/closed connection, on reconnection for some reason the prefix is added.

**Comment by Neborya** [comment]
Same issue here :(
I'm using thread-safe PECL DLL 4.0.5.2.
I don't use MemcachePool, just:
$memcache = new Memcache;
$memcache->connect("127.0.0.1", "11211");

Randomly a prefix 'localhost' will be added to stored keys. For example, after calling
$memcache->set("some_name", "some_value", false, 0);
the key can get saved as "localhostsome_name".

**Comment by Neborya** [comment]
@juongithub, have you managed to find a solution? I've tried different memcached server versions (up to 1.6.6) and different DLLs from @nono303, but the prefix still gets added at random, making it impossible to work under Windows.

**Comment by juongithub** [comment]
@Neborya did you try to empty  $_SERVER['SERVER_NAME'] just before trying to do anything with Memcache? For example:
```
$memcache = new Memcache;
$memcache->connect("127.0.0.1", "11211");
$_SERVER['SERVER_NAME'] = '';
$memcache->set("some_name", "some_value", false, 0);
```

**Comment by Neborya** [comment]
Trying it now, ty!
Still getting these prefix changes.
Is it enough to call `$_SERVER['SERVER_NAME'] = '';` once at the beginning of the script?

**Comment by juongithub** [comment]
Try in this way:
```
$_SERVER['SERVER_NAME'] = '';
$memcache = new Memcache;
$memcache->connect("127.0.0.1", "11211");
$memcache->set("some_name", "some_value", false, 0);
```

in my case it worked.

**Comment by Neborya** [comment]
Been testing for about an hour, got these prefix changes twice (which is acceptable and much less than before).
@juongithub, ty for you tip!
Do you know if this bug only occurs on Windows, and the lib is safe to use on the production server (under Linux)?

**Comment by juongithub** [comment]
This bug definitely occurs on Windows, as far as I know it does not occur on Linux (but I can't tell you for sure).

**Comment by nono303** [comment]
Hi,
The answer is somewhere in [get_session_key_prefix](https://github.com/websupport-sk/pecl-memcache/blob/NON_BLOCKING_IO_php7/php7/memcache.c#L323) and/or [get_key_prefix](https://github.com/websupport-sk/pecl-memcache/blob/NON_BLOCKING_IO_php7/php7/memcache.c#L393)

@Neborya are U testing in multi-vhost concurrency? with TS or NTS?
What are your php.ini values for keys: 
```
memcache.prefix_host_key
memcache.prefix_host_key_remove_www
memcache.prefix_host_key_remove_subdomain
memcache.prefix_static_key
```

**Comment by Neborya** [comment]
Hi!
Sorry for the late reply, there should not be multi-vhost concurrency (I am using almost unmodified XAMPP setup, and the scripts are launched one at a time through http://localhost/...). The DLL version is TS.

> What are your php.ini values for keys:
> ```
> memcache.prefix_host_key
> memcache.prefix_host_key_remove_www
> memcache.prefix_host_key_remove_subdomain
> memcache.prefix_static_key
> ```
I don't have these keys in my php.ini. @nono303 what values would you suggest for them?

**Comment by nono303** [comment]
hi @Neborya 
What does phpinfo said ?
like for me:
![image](https://user-images.githubusercontent.com/2869665/85308410-79915480-b4b1-11ea-9707-ad7575f28408.png)

**Comment by Neborya** [comment]
@nono303, I have exactly the same values in phpinfo.

**Comment by nono303** [comment]
So, if it's not a configuration issue, they might be a bug in [get_session_key_prefix](https://github.com/websupport-sk/pecl-memcache/blob/NON_BLOCKING_IO_php7/php7/memcache.c#L323) and/or [get_key_prefix](https://github.com/websupport-sk/pecl-memcache/blob/NON_BLOCKING_IO_php7/php7/memcache.c#L393)...

**Comment by juongithub** [comment]
I don't know if this can help, currently I'm not able to see the issue again, I'm using PHP 7.3.13.

**Comment by Neborya** [comment]
I still have it (without your fix `$_SERVER['SERVER_NAME'] = '';`) on PHP 7.4.

**Comment by juongithub** [comment]
@Neborya can you test PHP 7.3.13? Just for comparison

**Comment by Neborya** [comment]
@juongithub, sure, I just need to figure out how to replace my php version with 7.3.13 in XAMPP.

**Comment by rgagnon24** [comment]
+1 for this bug.  I see the same thing using php-pecl-memcache 4.0.5.2 for PHP 7.4
(php_memcache-4.0.5.2-7.4-ts-vc15-x64.zip)

Seemed random for a while, and still could be, but I also notice it when running XAMPP or straight out php-cgi... Seems to mostly occur if the connection to the server was lost for a abnormal disconnect... 

I have all the same memcache.XXXXX settings shown above as posted by @nono303 in https://github.com/websupport-sk/pecl-memcache/issues/56#issuecomment-647612228

**Comment by rgagnon24** [comment]
The problem seems to be in the macro MEMCACHE_G(prefix_host_key)

if you run ini_get('memcache.prefix_host_key'); you get back the STRING "0"
if you run ini_set('memcache.prefix_host_key', 0); the value still gets returned as a string "0"
but
if you run ini_set('memcache.prefix_host_key', false); the value comes back as a string "", and the problem goes away.

**Comment by nono303** [comment]
effectively, setting `memcache.prefix_host_key = "" ` & `memcache.session_prefix_host_key = ""` change value from **0** to _**no value**_
![image](https://user-images.githubusercontent.com/2869665/89495463-9e6b2e00-d7b8-11ea-8523-eb97c81b50d4.png)
![image](https://user-images.githubusercontent.com/2869665/89495469-a1661e80-d7b8-11ea-83dc-6292c1b6a8ca.png)
@Neborya, could you have a try please?

**Comment by rgagnon24** [comment]
After trying out the blank value for a while, it eventually stopped working.

So there is just something wrong with the zend library that is sometimes testing the "0" value and replying true, which then allows the code to capture the $_SERVER['SERVER_NAME'] as a prefix.  I don't know that code well enough to know if there is an event firing or something like a pointer being compared instead of a string, or if a string of any value is computed to be true, or what, but the issue is definitely in the code comparing the config value because the logic that uses it appears sound.

The ONLY working workaround I've found is to capture $_SERVER['SERVER_NAME'], then unset() it before creating the Memcache() object, adding servers to it, and then restoring the value to $_SERVER['SERVER_NAME'] so that other processes can use it.
```
$sn = $_SERVER['SERVER_NAME'] ?? null;
unset($_SERVER['SERVER_NAME']);
$mc = new \Memcache();
$mc->addServer('.....');
if ($sn) $_SERVER['SERVER_NAME'] = $sn;
```

This way, if the if comparison for prefix_host_key goes wrong, it will copy in a blank string as the prefix, resulting in a workaround that doesn't have a bad key.

**Comment by rgagnon24** [comment]
Could the definition of the config value just be wrong for php 7.4 compared with 7.3.x ??
Did Zend change something so there would be a needed change to:
https://github.com/websupport-sk/pecl-memcache/blob/NON_BLOCKING_IO_php7/php7/memcache.c#L300

**Comment by rgagnon24** [comment]
Ummm.... could this be the problem????
https://github.com/websupport-sk/pecl-memcache/blob/NON_BLOCKING_IO_php7/php7/memcache_pool.h#L424

```prefix_host_key``` is defined in the global struct as a ```char *```, but it is a boolean.  Should it maybe be defined there as ```zend_bool``` ?

**Comment by nono303** [comment]
Hi,
Here is a **NTS vs16/x64/avx** patched version for testing
```
--- "a/php7/memcache_pool.h"
+++ "b/php7/memcache_pool.h"
@@ -415,15 +415,15 @@ ZEND_BEGIN_MODULE_GLOBALS(memcache)
 	long compress_threshold;
 	long lock_timeout;
 	char *session_key_prefix;
-	char *session_prefix_host_key;
-	char *session_prefix_host_key_remove_www;
-	char *session_prefix_host_key_remove_subdomain;
+	zend_bool *session_prefix_host_key;
+	zend_bool *session_prefix_host_key_remove_www;
+	zend_bool *session_prefix_host_key_remove_subdomain;
 	char *session_prefix_static_key;
 	char *session_save_path;
 	char *key_prefix;
-	char *prefix_host_key;
-	char *prefix_host_key_remove_www;
-	char *prefix_host_key_remove_subdomain;
+	zend_bool *prefix_host_key;
+	zend_bool *prefix_host_key_remove_www;
+	zend_bool *prefix_host_key_remove_subdomain;
 	char *prefix_static_key;
 ZEND_END_MODULE_GLOBALS(memcache)
```
[php_memcache.zip](https://github.com/websupport-sk/pecl-memcache/files/5047284/php_memcache.zip)

**Comment by rgagnon24** [comment]
I dont thing those zend_bools are supposed to be pointers.

**Comment by TysonAndre** [comment]
> Ummm.... could this be the problem????
https://github.com/websupport-sk/pecl-memcache/blob/NON_BLOCKING_IO_php7/php7/memcache_pool.h#L424

Yeah, that's very likely to be it.
There's probably uninitialized bytes - a zend_bool uses 1 byte, and a pointer usually takes 4 or 8 bytes, so there's 3 or 7 uninitialized bytes, and code reading from those uninitialized bytes . Only one byte is modified by the OnUpdateBool implementation.

```c
/* Standard message handlers */
ZEND_API ZEND_INI_MH(OnUpdateBool) /* {{{ */
{
	zend_bool *p = (zend_bool *) ZEND_INI_GET_ADDR();
	*p = zend_ini_parse_bool(new_value);
	return SUCCESS;
}
/* }}} */
```

Similar to how `long lock_timeout` is a value, not a pointer, those should be `zend_bool`, not `zend_bool *`

```patch
--- "a/php7/memcache_pool.h"
+++ "b/php7/memcache_pool.h"
@@ -415,15 +415,15 @@ ZEND_BEGIN_MODULE_GLOBALS(memcache)
 	long compress_threshold;
 	long lock_timeout;
 	char *session_key_prefix;
-	char *session_prefix_host_key;
-	char *session_prefix_host_key_remove_www;
-	char *session_prefix_host_key_remove_subdomain;
+	zend_bool session_prefix_host_key;
+	zend_bool session_prefix_host_key_remove_www;
+	zend_bool session_prefix_host_key_remove_subdomain;
 	char *session_prefix_static_key;
 	char *session_save_path;
 	char *key_prefix;
-	char *prefix_host_key;
-	char *prefix_host_key_remove_www;
-	char *prefix_host_key_remove_subdomain;
+	zend_bool prefix_host_key;
+	zend_bool prefix_host_key_remove_www;
+	zend_bool prefix_host_key_remove_subdomain;
 	char *prefix_static_key;
 ZEND_END_MODULE_GLOBALS(memcache)

**Comment by nono303** [comment]
Hi,
Here is a NTS vs16/x64/avx version for testing with @TysonAndre patch (`zend_bool` not `zend_bool *`)
[php-7.4.x_memcache.zip](https://github.com/websupport-sk/pecl-memcache/files/5055925/php-7.4.x_memcache.zip)
_

**Comment by rgagnon24** [comment]
@nono303 Is is possible to get an TS version as well to test?  XAMPP and PhpED use Thread Safe interpreters.

**Comment by rgagnon24** [comment]
I'd like to note as well you may want to check out the "displayer" options when defining ini settings for an extension.  Those would cause the output in phpinfo() to show "On" or "Off" instead of the literal string from the ini file.

**Comment by nono303** [comment]
Here are **TS & NTS vs16/x64/avx** versions with phpinfo bool "displayer" option, for testing 

[php-7.4.x_memcache.zip](https://github.com/websupport-sk/pecl-memcache/files/5059269/php-7.4.x_memcache.zip)

```
--- a/php7/memcache.c
+++ b/php7/memcache.c
@@ -292,14 +292,14 @@ PHP_INI_BEGIN()
 	STD_PHP_INI_ENTRY("memcache.session_redundancy",	"2",			PHP_INI_ALL, OnUpdateRedundancy,	session_redundancy,	zend_memcache_globals,	memcache_globals)
 	STD_PHP_INI_ENTRY("memcache.compress_threshold",	"20000",		PHP_INI_ALL, OnUpdateCompressThreshold,	compress_threshold,	zend_memcache_globals,	memcache_globals)
 	STD_PHP_INI_ENTRY("memcache.lock_timeout",			"15",			PHP_INI_ALL, OnUpdateLockTimeout,		lock_timeout,		zend_memcache_globals,	memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.session_prefix_host_key",       			"0",			PHP_INI_ALL, OnUpdateBool, session_prefix_host_key, zend_memcache_globals, memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.session_prefix_host_key_remove_www",    	"1",			PHP_INI_ALL, OnUpdateBool, session_prefix_host_key_remove_www, zend_memcache_globals, memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.session_prefix_host_key_remove_subdomain",  "0",			PHP_INI_ALL, OnUpdateBool, session_prefix_host_key_remove_subdomain, zend_memcache_globals, memcache_globals)
+	STD_PHP_INI_BOOLEAN("memcache.session_prefix_host_key",       			"0",			PHP_INI_ALL, OnUpdateBool, session_prefix_host_key, zend_memcache_globals, memcache_globals)
+	STD_PHP_INI_BOOLEAN("memcache.session_prefix_host_key_remove_www",    	"1",			PHP_INI_ALL, OnUpdateBool, session_prefix_host_key_remove_www, zend_memcache_globals, memcache_globals)
+	STD_PHP_INI_BOOLEAN("memcache.session_prefix_host_key_remove_subdomain",  "0",			PHP_INI_ALL, OnUpdateBool, session_prefix_host_key_remove_subdomain, zend_memcache_globals, memcache_globals)
 	STD_PHP_INI_ENTRY("memcache.session_prefix_static_key",         		NULL,			PHP_INI_ALL, OnUpdatePrefixStaticKey, session_prefix_static_key, zend_memcache_globals, memcache_globals)
 	STD_PHP_INI_ENTRY("memcache.session_save_path",         				NULL,			PHP_INI_ALL, OnUpdateString, session_save_path, zend_memcache_globals, memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.prefix_host_key",       					"0",			PHP_INI_ALL, OnUpdateBool, prefix_host_key, zend_memcache_globals, memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.prefix_host_key_remove_www",    			"1",			PHP_INI_ALL, OnUpdateBool, prefix_host_key_remove_www, zend_memcache_globals, memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.prefix_host_key_remove_subdomain",  		"0",			PHP_INI_ALL, OnUpdateBool, prefix_host_key_remove_subdomain, zend_memcache_globals, memcache_globals)
+	STD_PHP_INI_BOOLEAN("memcache.prefix_host_key",       					"0",			PHP_INI_ALL, OnUpdateBool, prefix_host_key, zend_memcache_globals, memcache_globals)
+	STD_PHP_INI_BOOLEAN("memcache.prefix_host_key_remove_www",    			"1",			PHP_INI_ALL, OnUpdateBool, prefix_host_key_remove_www, zend_memcache_globals, memcache_globals)
+	STD_PHP_INI_BOOLEAN("memcache.prefix_host_key_remove_subdomain",  		"0",			PHP_INI_ALL, OnUpdateBool, prefix_host_key_remove_subdomain, zend_memcache_globals, memcache_globals)
 	STD_PHP_INI_ENTRY("memcache.prefix_static_key",         				NULL,			PHP_INI_ALL, OnUpdatePrefixStaticKey, prefix_static_key, zend_memcache_globals, memcache_globals)
 PHP_INI_END()
 /* }}} */
diff --git a/php7/memcache_pool.h b/php7/memcache_pool.h
index 2382aa8..c2669c5 100644
--- a/php7/memcache_pool.h
+++ b/php7/memcache_pool.h
@@ -415,15 +415,15 @@ ZEND_BEGIN_MODULE_GLOBALS(memcache)
 	long compress_threshold;
 	long lock_timeout;
 	char *session_key_prefix;
-	char *session_prefix_host_key;
-	char *session_prefix_host_key_remove_www;
-	char *session_prefix_host_key_remove_subdomain;
+	zend_bool session_prefix_host_key;
+	zend_bool session_prefix_host_key_remove_www;
+	zend_bool session_prefix_host_key_remove_subdomain;
 	char *session_prefix_static_key;
 	char *session_save_path;
 	char *key_prefix;
-	char *prefix_host_key;
-	char *prefix_host_key_remove_www;
-	char *prefix_host_key_remove_subdomain;
+	zend_bool prefix_host_key;
+	zend_bool prefix_host_key_remove_www;
+	zend_bool prefix_host_key_remove_subdomain;
 	char *prefix_static_key;
 ZEND_END_MODULE_GLOBALS(memcache)
```
![image](https://user-images.githubusercontent.com/2869665/89943915-48b2dd80-dc1f-11ea-8e34-481330940652.png)

**Comment by rgagnon24** [comment]
I guess I'm out of luck for some reason... I get this error trying to start XAMPP with the new dll:
Can't load module 'C:\\xampp\\php\\ext\\php_memcache.dll' as it's linked with 14.27, but the core is linked with 14.16

**Comment by nono303** [comment]
14.27: vs16 / 14.16: vc15 ;)
**TS & NTS vc15/x64/avx**: [php-7.4.x_memcache.zip](https://github.com/websupport-sk/pecl-memcache/files/5062472/php-7.4.x_memcache.zip)

**Comment by rgagnon24** [comment]
THANKS, @nono303 That seems to work OK.

I ran a pile of tests with the 4.0.5.2 TS 64-bit DLL from the pecl download site and got a LOT of bad results where "localhost" was prefixed onto the keys.

The version you uploaded in https://github.com/websupport-sk/pecl-memcache/issues/56#issuecomment-672826394 works like it should, 100% without prefixing the hostname when the ini is set to false (rather than random like before).

I also tested by ENABLING the prefix_host_key and 100% of the messaged HAD the prefix key as expected.

I think this one might be solved.

**Comment by nono303** [comment]
@rgagnon24 Thanks for testing!
You're right for `allow_failover`, it must be `zend_bool`
```
memcache.c:     STD_PHP_INI_ENTRY("memcache.allow_failover",            "1",                    PHP_INI_ALL, OnUpdateLong,                      allow_failover, zend_memcache_globals,  memcache_globals)
memcache.c:     if (!MEMCACHE_G(allow_failover) || request->failed_servers.len >= MEMCACHE_G(max_failover_attempts)) {
memcache_pool.c:        if (!mmc_server_valid(mmc) && MEMCACHE_G(allow_failover)) {
memcache_pool.c:        if (MEMCACHE_G(allow_failover) && request->failed_index < MEMCACHE_G(max_failover_attempts) && request->failed_servers.len < pool->num_servers) {
memcache_pool.h:        long allow_failover;
```
I also cleaned view of `PHP_INI_BEGIN` funtion and checked that all others were well typed (that's now seems to be the case)

Zip file contain all dll versions for PHP 7.4 and the final(?) patch. [php-7.4.x_memcache.zip](https://github.com/websupport-sk/pecl-memcache/files/5067281/php-7.4.x_memcache.zip)

I let you verify  and update https://github.com/websupport-sk/pecl-memcache/pull/71 if you want.
I'll push them to https://github.com/nono303/PHP7-memcache-dll when your PR will be closed and tag will be updated (don't like version with short-tag ^^)

**Comment by rgagnon24** [comment]
I can confirm that the version uploaded in https://github.com/websupport-sk/pecl-memcache/issues/56#issuecomment-673270717 works the same as the previous update in regards to the prefix problem being resolved.  I only have one server to play with, so I cannot comment on the allow_failover since I didn't test it, however I do see that the phpinfo() output did properly change from showing "1" to showing "On" as the state of that ini setting.

**Comment by nono303** [comment]
all version released with tag [4.0.5.2-pr71](https://github.com/nono303/PHP7-memcache-dll/tree/4.0.5.2-pr71)

**Comment by thesource93** [comment]
this seems to be fixed,closing
